### ISSUE TITLE
Project '..' targets 'net5.0'. It cannot be referenced by a project that targets '.NETCoreApp,Version=v5.0'

### ISSUE THREAD

**Issue by hypeartist** [dotnet/sdk#12559]
While trying to reference a NET5.0 class library project from a NET5.0 WinForm project I get the error stated in the title:

> Project 'classlib.csproj' targets 'net5.0'. It cannot be referenced by a project that targets '.NETCoreApp,Version=v5.0'

**Comment by ViktorHofer** [comment]
@dsplaisted I transferred this over to dotnet/sdk.

**Comment by dsplaisted** [comment]
@ansarizafar Likely you need to update the version of the .NET SDK you are using.  @DustinCampbell

**Comment by wouldBeNerd** [comment]
@ansarizafar for vscode I changed the version in my .csproj file from `<TargetFramework>net5.0</TargetFramework>` to `<TargetFramework>net4.8</TargetFramework>`
then I closed and reopened vscode on my project folder and the error was no more. 

assuming that you are also using the extension `C# for Visual Studio Code (powered by OmniSharp)`

**Comment by mrj001** [comment]
I'm getting this same error.  These are the versions I'm using:
```
$ dotnet nuget
NuGet Command Line 5.8.0.4
```

```
$ dotnet build
Microsoft (R) Build Engine version 16.8.0-preview-20451-02+51a1071f8 for .NET
```

and
```
$ dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.1.20454.5
 Commit:    53a032d031

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /Users/mj/Documents/develop/runtime/.dotnet/sdk/5.0.100-rc.1.20454.5/

Host (useful for support):
  Version: 5.0.0-rc.1.20451.14
  Commit:  38017c3935

.NET SDKs installed:
  5.0.100-rc.1.20454.5 [/Users/mj/Documents/develop/runtime/.dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.0-rc.1.20451.17 [/Users/mj/Documents/develop/runtime/.dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.0-rc.1.20451.14 [/Users/mj/Documents/develop/runtime/.dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download

```

**Comment by ViktorHofer** [comment]
cc @nkolev92

**Comment by nkolev92** [comment]
We did some fixes in https://github.com/NuGet/NuGet.Client/commit/935b1d2f5bef8d0d78a737190846b72f0bae30e9 and https://github.com/dotnet/msbuild/commit/79ab985eb5b6552d96d237d4179237b4b850ca64. 

I'm not sure which was the first version of the SDK that had both these fixes though.

**Comment by robertodalmonte** [comment]
I experienced the same problem (now solved) when I tried to upgrade a solution (everything set to netcoreapp3.1) to everything set to net5.0.
In my case, I had a gRPC service, with a reference to a WPF project. Everything worked nicely...until I tried to upgrade the solution:
Error NU1201 Project OneClick.Document is not compatible with net5.0.
The message is very misleading...in fact, you can have everything set to net5.0. It was a bit tricky in my case, since I had a gRPC service who had a reference on a WPF project. The solution is somewhat weird:

<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>net5.0-windows</TargetFramework>
  </PropertyGroup>

[......]

</Project>

On the gRPC service, since you're referencing a WPF project, you need to set

    <TargetFramework>net5.0-windows</TargetFramework>
instead of
    <TargetFramework>net5.0</TargetFramework>
Nonetheless, at the same time, the Project Sdk must be Microsoft.NET.Sdk.Web.

While trying to understand this problem I created a new gRPC service (net5.0 by default), then I created a new WPF project (netcoreapp3.1 by default) and when I tried to reference it, I got the right suggestion; you don't get it if the gRPC service TargetFramework is set to netcoreapp3.1, which was my situation originally.
Now I have everything set to net5.0 (or net5.0-windows) and it works like a charm.

**Comment by robertodalmonte** [comment]
Me again...
I do have a minor side effect: the publish would fail:
C:\Program Files\dotnet\sdk\5.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(241,5): Error NETSDK1005: Assets file 'S:\Codice\2021\InformedConsentServer\obj\project.assets.json' doesn't have a target for 'net5.0'. Ensure that restore has run and that you have included 'net5.0' in the TargetFrameworks for your project.

I can build the solution and manually copy the files, though.
bye

**Comment by eviathan** [comment]
Is there a reason that this is still a problem?

**Comment by robertodalmonte** [comment]
> 
> 
> Is there a reason that this is still a problem?

Hello eviathan,
if you're referring to the publish fail, then yes, to my knowledge it is still an issue.
If you're referring to the general issue of this topic, I was able to solve it, as I explained in the first of my 2 messages. I do not know if my case is a peculiar lucky case or if that is the solution to the issue...

**Comment by dsplaisted** [comment]
@robertodalmonte Can you [share a binlog](https://aka.ms/binlog) of the failed publish operation?  Thanks.

**Comment by robertodalmonte** [comment]
[binlog.zip](https://github.com/dotnet/sdk/files/5687465/binlog.zip)
Hopefully, this is what you're looking for...

This is from Visual Studio
14/12/2020 09:29:43
System.AggregateException: One or more errors occurred. ---> Microsoft.WebTools.Shared.Exceptions.WebToolsException: Build failed. Check the Output window for more details.
   --- End of inner exception stack trace ---
---> (Inner Exception #0) Microsoft.WebTools.Shared.Exceptions.WebToolsException: Build failed. Check the Output window for more details.<---

Microsoft.WebTools.Shared.Exceptions.WebToolsException: Build failed. Check the Output window for more details.

===================

**Comment by dsplaisted** [comment]
Does it fail if you publish from the command line, or just from within Visual Studio?

**Comment by robertodalmonte** [comment]
I use Visual Studio 2019 Community Edition, and it fails.
Just now I tried in Visual Studio Code, from terminal:
dotnet publish --configuration Release
and it worked nicely
![Screenshot (29)](https://user-images.githubusercontent.com/24954850/102063295-22cc5e00-3df6-11eb-8030-4d961703ff50.png)

**Comment by dsplaisted** [comment]
@vijayrkn Could you look at @robertodalmonte's issue here with publish from VS?  It looks like it's not restoring for the right RuntimeIdentifier or something.

**Comment by goko1** [comment]
Updating Visual Studio from Visual Studio Installer solve my similar error.

**Comment by Ryanman** [comment]
Moving from 16.7.7 to 16.8.3 solved this error for me.

**Comment by DanPFW** [comment]
> Moving from 16.7.7 to 16.8.3 solved this error for me.

I have this same problem on 16.8.3.  Did not happen before I installed .net5.0

**Comment by thebhef** [comment]
I'm experiencing this issue in 16.7.13 after rolling back from 16.9.2 due to https://github.com/dotnet/roslyn/issues/52099 
The build is fine using `dotnet build` but VS just spits out a lot of these errors:

```
error NETSDK1005: Assets file '<PROJECT>\project.assets.json' doesn't have a target for '.NETFramework,Version=v5.0'. Ensure that restore has run and that you have included 'net50' in the TargetFrameworks for your project.
```

No combination of clean/rebuild operations has helped. This build works fine on 16.8.x, but I don't see that I have any way to go back to a 16.8 version

**Comment by dgm3333** [comment]
I've also forked UVtools and just installed 
dotnet-sdk-5.0.201-win-x64.exe  and 
ndp48-devpack-enu.exe

But the build failed from within VS.
I tried doing a global replace of <TargetFramework>net4.8</TargetFramework> to <TargetFramework>net5.0</TargetFramework> which found 3 instances and meant I got further in the build, but still failed.
When I use powershell to run
dotnet publish --configuration Release
I get the following error (but not sure where the UVtools.GUI.csproj target framework is set as I've never coded .NET before)
C:\Program Files\dotnet\sdk\5.0.201\Microsoft.Common.CurrentVersion.targets(1720,5): error : Project '..\UVtools.Core\UVtools.Core.csproj' targets 'net5.0'. It cannot be referenced by a project that targets '.NETFramework,Version=v4.8'. [C:\UVtools\UVtools.GUI\UVtools.GUI.csproj]
